batchflow/models/torch/base_batchsize_mixin.py

```diff
@@ -0,0 +1,336 @@
+
+                next(generator)
+                generator.send(batch_size[-1])
+                new_batch_size, exit = generator.send(consumed_memory)
+                batch_size.append(new_batch_size)
+            except RuntimeError as exception:
+                if self.is_oom_error(exception):
+                    next(generator)
+                    generator.send(batch_size[-1])
+                    new_batch_size, exit = generator.send(max_memory * 2)
+                    batch_size.append(new_batch_size)
+                else:
+                    raise # some other error not memory related
+
+            self.garbage_collection_cuda()
+            if exit or count >= max_iters:
+                break
+
+        return batch_size[-1]
+
+
+    def _compute_optimal_batch_size_predictive(self, method='train', max_memory=90,
+                                               inputs=None, targets=None, pbar='n',
+                                               max_iters=16, start_batch_size=4,
+                                               delta_batch_size=4, max_batch_size=128,
+                                               n=20, frequency=0.05, time_threshold=3,
+                                               tail_size=20, std_threshold=0.1):
+        """ Compute optimal batch size for training/inference to maximize GPU memory usage.
+        Works by using `train`/`predict` with different batch sizes, and measuring how much memory is taken.
+        Then, we solve the system of `measured_memory = batch_size * item_size + model_size + eps` equations for both
+        `item_size` and `model_size`.
+
+        For stable measurements, we make `n` iterations of `train`/`predict`, until the memory consumption stabilizes.
+        """
+        #pylint: disable=consider-iterating-dictionary
+        table = {}
+        batch_size = start_batch_size
+        for _ in Notifier(pbar)(range(max_iters)):
+            info = self.get_memory_utilization(batch_size, method=method,
+                                               inputs=inputs, targets=targets, n=n, frequency=frequency,
+                                               time_threshold=time_threshold,
+                                               tail_size=tail_size, std_threshold=std_threshold)
+            table[batch_size] = info
+
+            # Exit condition
+            batch_size += delta_batch_size
+            if info['memory'] > max_memory or batch_size > max_batch_size:
+                break
+
+        # Make and solve a system of equations for `item_size`, `model_size`
+        matrix = np.array([[batch_size, 1] for batch_size in table.keys()])
+        vector = np.array([value['memory'] for value in table.values()])
+        item_size, model_size = np.dot(np.linalg.pinv(matrix), vector)
+
+        # Compute the `batch_size` to use up to `max_memory`
+        optimal_batch_size = (max_memory - model_size) / item_size
+        optimal_batch_size = int(optimal_batch_size)
```

Comment:
```suggestion
        optimal_batch_size = (max_memory - model_size) // item_size
```
If you wanted floor rounding

Reply:
In your implementation optimal_batch_size will be float, because both numenator and denominator are floats. So anyway I have to convert it to int.

Reply:
You are right, my mistake